sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -176,17 +182,41 @@ function calculateQueryParameters(
   return result;
 }
 
+function simpleParseQueryParams(queryString: string): Array<[string, string]> {
```

Comment:
How about using [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)?

Reply:
This causes problems (per the comment before we call this method), since it decodes special characters when you read them in values, but since we have to join the query params manually (to avoid encoding comma), it would stay unencoded and no longer be valid.